client/my-sites/stats/stats-notices/all-notice-definitions.ts

```diff
@@ -130,4 +103,39 @@ const ALL_STATS_NOTICES: StatsNoticeType[] = [
 	},
 ];
 
+function shouldShowCommercialSiteUpgradeNotice( {
+	isOdysseyStats,
+	isWpcom,
+	isVip,
+	isP2,
+	isOwnedByTeam51,
+	hasPaidStats,
+	isSiteJetpackNotAtomic,
+	isCommercial,
+	hasPWYWPlanOnly,
+}: StatsNoticeProps ) {
+	const showUpgradeNoticeForWpcomSites = isWpcom && ! isP2 && ! isOwnedByTeam51;
+	const showUpgradeNoticeOnOdyssey = isOdysseyStats;
+	const showUpgradeNoticeForJetpackNotAtomic = isSiteJetpackNotAtomic;
+
+	// Test specific to commercial self-hosted sites with PWYW plans.
+	// They should see the upgrade notice!
+	if ( showUpgradeNoticeOnOdyssey || showUpgradeNoticeForJetpackNotAtomic ) {
+		if ( isCommercial && hasPWYWPlanOnly ) {
+			return true;
+		}
+	}
```

Comment:
I understand this might be written intentionally in two conditions but for the sake of simplicity, could we merge them to one?